### PR TITLE
cli: refactor data printing, print on create/update

### PR DESCRIPTION
**Pull request by tommilligan**
This PR:

- refactors the collection of `print_*_table` functions into a struct, `Printer`, and trait, `DisplayTable`.
  - also extends the `Iterator` trait to allow easy printing of iterables of resources
- adds printing functionality to the resources returned in the `create` and `update` subcommands

This is a breaking change. Users updating to a new version of the cli will need to update their usage. If desired, I could leave the old `-o/--output` flags in, and somehow shim them in (overriding the global output flag preference). This would make it a back-compatible change.

New sample usage:

```bash
$ re -ojson create source org1/tmp | jq
W TLS certificate verification is disabled. Do NOT use this over an insecure network.
I New source `org1/tmp` [id: 7fde97d466fcc74e] created successfully
{
  "id": "7fde97d466fcc74e",
  "owner": "org1",
  "name": "tmp",
  "title": "",
  "description": "",
  "language": "en",
  "should_translate": false,
  "created_at": "2021-07-02T13:43:22.020Z",
  "updated_at": "2021-07-02T13:43:22.020Z"
}
```

Open to bikeshedding on the trait names :fearful:

### CODE REVIEW

**Comment by jcalero** [comment]
Very cool! I don't feel confident judging the Rust, but functionality-wise it's great! Nice work factoring out the printer logic.

**Comment by tommilligan** [comment]
I'll wait on a final decision on the cli flags we want before fixing up the tests that are failing.

**Review by mcobzarenco — APPROVED** [review]
Defo super useful in scripts + very nice refactor of how we print resources re: removing a lot of duplication :100:  

Had more of a question than an opinion on whether we should make the argument global, but I'm easy.

File: cli/src/args.rs
Comment (tommilligan) [reply cli/src/args.rs:55]: As discussed in slack, we're going to move to a global flag, as we think user readable/data output is a useful paradigm in the vast majority of routes: https://reinfer.slack.com/archives/C0APLNE23/p1625243630379900?thread_ts=1625234925.358800&cid=C0APLNE23

> I think having table/json outputs for the following commands would still be useful:
> - re config add/current/ls/use
>
> The ones I explicitly think it will be never useful in are:
> - re completion
> - re help
>
> which also already don't need args such as --context, --token or --proxy which are other top level args
>
> So I think I am leaning back to global flag

**Pull request merged**
